docs(js): Add sourcemaps explain to source maps troubleshooting guide (#5076)
getsentry/sentry-docs: src/platforms/javascript/common/sourcemaps/troubleshooting_js.mdx

sidebar_order: 8
---

> Source maps can sometimes be tricky to get going. If you are having trouble, you can try using our verification tool inside `sentry-cli`, or go through the steps listed below yourself. Please keep in mind, that the tool mentioned above is in it’s early stage, and we are actively working to improve it.

Reviewer: ```suggestion
Source maps can sometimes be tricky to get going. If you're having trouble, you can try using our verification tool inside `sentry-cli`, or go through the steps listed below yourself. Please keep in mind, that the `sentry-cli` is a work-in-progress, and we are actively working to improve it.
```

To use the automated process, [install and configure](/product/cli/) `sentry-cli`. Once completed, use the `sourcemaps explain` command, by calling it with the event ID that can be found at the top header of the issue page, eg. `Event fdfd1337e3964cd6a4d11f35357a4c43 | JSON (42.0 KiB)`.

```shell
sentry-cli sourcemaps explain fdfd1337e3964cd6a4d11f35357a4c43
```

This should provide you with some useful hints about what exactly went wrong with the source maps setup process.

> If that is not the case, let us know, so that we can make it better.

Reviewer: ```suggestion
If it doesn't, let us know so we can make it better.
```

<PlatformSection notSupported={["javascript.electron"]}>

## Verify a release is configured in your SDK

For uploaded source maps to be located and applied, the release needs to be created by the CLI or API (and the correct artifacts uploaded with it), and the name of that newly-created release needs to be specified in your SDK configuration.

> To verify this, open up the issue from the Sentry UI and check if the release is configured. If it says "_not configured_" or "_N/A_" next to **Release** on the right hand side of the screen (or if you do not see a `release` tag in the list of tags), you will need to go back and [tag your errors](../../configuration/releases/). If this is properly set up you will see "Release: my_example_release".

Author: Oh, I thought it was other way around. Thanks!

sidebar_order: 8
---

Source maps can sometimes be tricky to get going. If you're having trouble, you can try using our verification tool inside `sentry-cli`, or go through the steps listed below yourself. Please keep in mind, that the `sentry-cli` is a work-in-progress, and we are actively working to improve it.

To use the automated verification process, [install and configure](/product/cli/) `sentry-cli`. Once completed, use the `sourcemaps explain` command, by calling it with the event ID that can be found at the top of the **Issue Details** page in [sentry.io](https://sentry.io). For example, "Event fdfd1337e3964cd6a4d11f35357a4c43 | JSON (42.0 KiB)".

```shell
sentry-cli sourcemaps explain fdfd1337e3964cd6a4d11f35357a4c43
```

> This should provide you with some useful hints about what exactly went wrong with the source maps setup process.
> If it doesn't, let us know so we can make it better.

Author: ```suggestion
This should provide you with some useful hints about what exactly went wrong with the source maps setup process.
If it doesn’t, let us know so we can make it better.
```

//# sourceMappingURL=script.min.js.map
```

> An example `sentry-cli` command to upload these files correctly would look like this (assuming you are in the `/scripts` directory, running your web server from one directory higher, which is why we are using the `--url-prefix` option):

Author: ```suggestion
An example `sentry-cli` command to upload these files correctly would look like this (assuming you’re in the `/scripts` directory, running your web server from one directory higher, which is why we’re using the `--url-prefix` option):
```

npm install -g source-map
```

> Then, write a script that reads your source map file and tests a mapping. Here is an example:

Author: ```suggestion
Then, write a script that reads your source map file and tests a mapping. Here’s an example:
```

The Sentry API currently only works with source maps and source files that are uploaded as plain text (UTF-8 encoded). If the files are uploaded in a compressed format (for example, gzip), they will be not be interpreted correctly.

> This sometimes occurs with build scripts and plugins that produce pre-compressed minified files. For example, Webpack’s [compression plugin](https://github.com/webpack/compression-webpack-plugin). You will need to disable such plugins and perform the compression _after_ the generated source maps/source files have been uploaded to Sentry.

Author: ```suggestion
This sometimes occurs with build scripts and plugins that produce pre-compressed minified files. For example, Webpack’s [compression plugin](https://github.com/webpack/compression-webpack-plugin). You’ll need to disable such plugins and perform the compression _after_ the generated source maps/source files have been uploaded to Sentry.
```
